### PR TITLE
fix: code quality issues

### PR DESCRIPTION
**Pull request by withshubh**
#### Description
This PR fixes a few issues that were affecting the code quality.

#### Summary of changes
- Remove unnecessary use of comprehension
- Remove unnecessary comprehension
- Use literal syntax instead of function calls to create data structure
- Pass string format arguments as logging method parameters
- Remove unused imports
- Remove unnecessary generator
- Refactor `if` expression

### CODE REVIEW

**Comment by withshubh** [comment]
Hwy @pkkid :wave: Please review this PR! :sparkles:

File: plexapi/base.py [hunk 1]
```diff
@@ -501,7 +501,7 @@ def delete(self):
             return self._server.query(self.key, method=self._server._session.delete)
         except BadRequest:  # pragma: no cover
             log.error('Failed to delete %s. This could be because you '
-                'havnt allowed items to be deleted' % self.key)
+                'havnt allowed items to be deleted', self.key)
```
Comment (jjlawren) [inline plexapi/base.py:504]:
```suggestion
                'have not allowed items to be deleted', self.key)
```

**Review by jjlawren — CHANGES_REQUESTED** [review]
All good changes. Might as well fix this spelling mistake while we're at it.

**Comment by withshubh** [comment]
@jjlawren fixed :sparkles:

**Review by jjlawren — APPROVED** [review]

**Comment by jjlawren** [comment]
Looks like a recent merge led to a conflict. Can you resolve?

**Comment by withshubh** [comment]
@jjlawren done!

File: tests/test_audio.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # -*- coding: utf-8 -*-
+
```
Comment (jjlawren) [inline tests/test_audio.py:2]:
```suggestion
```

**Review by jjlawren — APPROVED** [review]

**Comment by withshubh** [comment]
@jjlawren I would like to inform you that I ran [DeepSource](https://deepsource.io/) analyzer on my fork of this repository to detect these issues. Have a look at the issues caught in this repository by DeepSource [here](https://deepsource.io/gh/withshubh/python-plexapi/issues/?category=recommended).

DeepSource is a code review automation tool that detects code quality issues and helps you to automatically fix some of them. You can use DeepSource to track test coverage, Detect problems in Dockerfiles, etc. in addition to detecting issues in code.

Can I submit a new PR with the configuration file which I used to configure the analysis?
You can merge the configuration file to continuously analyze the repo and fix code quality issues.

**Pull request merged**